src/Compilers/CSharp/Portable/Lowering/SyntheticBoundNodeFactory.cs

```diff
@@ -836,9 +836,11 @@ public BoundSequence Sequence(BoundExpression[] sideEffects, BoundExpression res
             return new BoundSequence(Syntax, ImmutableArray<LocalSymbol>.Empty, sideEffects.AsImmutableOrNull(), result, type ?? result.Type) { WasCompilerGenerated = true };
         }
 
-        public BoundSequence Sequence(ImmutableArray<LocalSymbol> locals, ImmutableArray<BoundExpression> sideEffects, BoundExpression result)
+        public BoundExpression Sequence(ImmutableArray<LocalSymbol> locals, ImmutableArray<BoundExpression> sideEffects, BoundExpression result)
         {
-            return new BoundSequence(Syntax, locals, sideEffects, result, result.Type) { WasCompilerGenerated = true };
+            return
+                locals.IsDefaultOrEmpty && sideEffects.IsDefaultOrEmpty ? result :
+                new BoundSequence(Syntax, locals, sideEffects, result, result.Type) { WasCompilerGenerated = true };
```

Comment:
Consider indenting this line so it's clear it's part of a ternary, and not just being && or || with the previous line.